### DISCUSSION TITLE
Feature request: Ability to export records as JSON / CSV or XLS

### DISCUSSION THREAD

**Discussion by quornik** [payloadcms/payload#1557] [Feature Requests & Ideas]
Since Payload is using Mongo as database, it would be great to be able to download collection records as CSV, the same way Mongo Compass or CLI is able to do.

**Comment by jmikrut** [comment]
Absolutely. Marking as roadmap item priority 3. Let's do it!

**Reply by nguoiviettinhca** [reply]
Importing CSV files to a collection would be nice also

**Comment by itsjavi** [comment]
Import and Export JSON documents would be nice, also to have it available via CLI for automation.

What I wish to be able to do:

- Export all collections to JSON files
   - Ability to use a document transformer function to modify the documents on export
- Import collections from JSON files
   - Ability to use a document loader function to modify the documents on import (e.g. in case we have a non-standard format),
- Export/Import commands available via CLI for automation and CIs

**Comment by rafaell-lycan** [comment]
Hey all 👋

I'm looking for a similar feature to transfer (export/import) data from one CMS to another 🙏

**Comment by GermanJablo** [comment]
Closing as duplicate of https://github.com/payloadcms/payload/discussions/4309

Also for the record, this feature is already being worked on! 🥳
https://github.com/payloadcms/payload/pull/10795
